dbms/src/DataStreams/IBlockInputStream.cpp

```diff
@@ -247,6 +247,26 @@ void IBlockInputStream::checkQuota(Block & block)
     }
 }
 
+static void limitProgressingSpeed(size_t total_progress_size, size_t max_speed_in_seconds, UInt64 total_elapsed_microseconds)
```

Comment:
Why don't use Throttler here? The logic looks the same.

Reply:
Because `Throttler` accepts different bytes value, this will cause the last processed bytes value to be stored in `IBlockInputStream`